app/feature/feature-changeaddress/src/main/kotlin/com/hedvig/android/feature/changeaddress/destination/selecthousingtype/SelectHousingTypeViewModel.kt

```diff
@@ -0,0 +1,136 @@
+    presenter = SelectHousingTypePresenter(changeAddressRepository),
+  )
+
+internal class SelectHousingTypePresenter(private val changeAddressRepository: ChangeAddressRepository) :
+  MoleculePresenter<SelectHousingTypeEvent, SelectHousingTypeUiState> {
+  @Composable
+  override fun MoleculePresenterScope<SelectHousingTypeEvent>.present(
+    lastState: SelectHousingTypeUiState,
+  ): SelectHousingTypeUiState {
+    var dataLoadIteration by remember {
+      mutableIntStateOf(0)
+    }
+
+    var currentState by remember {
+      mutableStateOf(lastState)
+    }
+
+    CollectEvents { event ->
+      when (event) {
+        DismissErrorDialog -> currentState = currentState.copy(errorMessage = null)
+
+        DismissHousingTypeErrorDialog -> currentState = currentState.copy(errorMessageRes = null)
+
+        is SelectHousingType -> {
+          currentState = currentState.copy(housingType = ValidatedInput(input = event.housingType))
+        }
+
+        SubmitHousingType -> {
+          if (!currentState.housingType.isPresent) {
+            currentState = currentState.copy(
+              errorMessageRes = R.string.CHANGE_ADDRESS_HOUSING_TYPE_ERROR,
+            )
+          } else {
+            dataLoadIteration++
+          }
+        }
+
+        ClearNavigationParameters -> currentState = currentState.copy(navigationParameters = null)
+      }
+    }
+
+    LaunchedEffect(dataLoadIteration) {
+      if (dataLoadIteration > 0) {
+        currentState = currentState.copy(isLoading = true)
+        changeAddressRepository.createMoveIntent().fold(
+          ifLeft = { error ->
+            currentState = currentState.copy(
+              isLoading = false,
+              errorMessage = error.message,
+            )
+          },
+          ifRight = { moveIntent ->
+            val parameters = SelectHousingTypeParameters(
+              minDate = moveIntent.movingDateRange.start,
+              maxDate = moveIntent.movingDateRange.endInclusive,
+              moveIntentId = moveIntent.id.id,
+              suggestedNumberInsured = moveIntent.suggestedNumberInsured.toString(),
+              moveFromAddressId = moveIntent.currentHomeAddresses.firstOrNull()?.id,
+              extraBuildingTypes = moveIntent.extraBuildingTypes,
+              isEligibleForStudent = moveIntent.isApartmentAvailableforStudent == true &&
+                currentState.housingType.input != HousingType.VILLA,
+              maxSquareMeters = when (currentState.housingType.input) {
+                HousingType.APARTMENT_RENT,
+                HousingType.APARTMENT_OWN,
+                -> moveIntent.maxApartmentSquareMeters
+
+                HousingType.VILLA -> moveIntent.maxHouseSquareMeters
+                null -> null
+              },
+              maxNumberCoInsured = when (currentState.housingType.input) { // todo: is this number of coInsured or just Insured (incl.main insured?) bc later it's used as if it is the later one.
```

Comment:
I don't know myself either actually, probably worth running with a sample memeber to see what it really is to make sure we're not misusing it

Reply:
I did, it seems to be overall for main+co-insured. 
Also, Sladan says in iOs they can go as high as up to 6 co-insured. I wonder how it corresponds with this maxNumber